File: agroguard/frontend/src/types/areaIntelligence.ts
export type SoilReport = {
  state: string;
  district: string;
  soil_type: string;
  ph_level: string;
  nitrogen_level: string;
  phosphorus_level: string;
  potassium_level: string;
  organic_matter: string;
  moisture_retention: string;
  drainage_capacity: string;
};

export type CropRecommendation = {
  crop_name: string;
  scientific_name: string;
  crop_image: string;
  growing_season: string;
  water_requirement: string;
  expected_yield: string;
  suitable_soil_type: string;
  ideal_temperature_range: string;
  rainfall_requirement: string;
  seed_rate_per_acre: string;
  average_growth_duration: string;
  fertilizer_recommendation: string;
  pest_disease_risk_level: string;
  market_demand_indicator: string;
  planting_method: string;
  irrigation_schedule: string;
  irrigation_method: string;
  common_diseases: string[];
  disease_prevention: string;
  market_price_range: string;
};

export type WaterGuidance = {
  crop_name: string;
  water_requirement: string;
  recommended_method: string;
};

export type AreaIntelligenceResponse = {
  state: string;
  district: string;
  soil_report: SoilReport;
  soil_improvement_recommendations: string[];
  recommended_crops: CropRecommendation[];
  water_guidance: WaterGuidance[];
  generated_at: string;
};

export type DetectDistrictResponse = {
  state: string;
  district: string;
  latitude: number;
  longitude: number;
  source: string;
};

File: ShetVaidya/backend/app/schemas/area_intelligence.py
from datetime import datetime
from typing import List

from pydantic import BaseModel, Field


class DetectDistrictRequest(BaseModel):
    latitude: float = Field(..., ge=-90, le=90)
    longitude: float = Field(..., ge=-180, le=180)


class DetectDistrictResponse(BaseModel):
    state: str
    district: str
    latitude: float
    longitude: float
    source: str


class DistrictListResponse(BaseModel):
    state: str
    districts: List[str]


class SoilReportOut(BaseModel):
    state: str
    district: str
    soil_type: str
    ph_level: str
    nitrogen_level: str
    phosphorus_level: str
    potassium_level: str
    organic_matter: str
    moisture_retention: str
    drainage_capacity: str


class CropRecommendationOut(BaseModel):
    crop_name: str
    scientific_name: str
    crop_image: str
    growing_season: str
    water_requirement: str
    expected_yield: str
    suitable_soil_type: str
    ideal_temperature_range: str
    rainfall_requirement: str
    seed_rate_per_acre: str
    average_growth_duration: str
    fertilizer_recommendation: str
    pest_disease_risk_level: str
    market_demand_indicator: str
    planting_method: str
    irrigation_schedule: str
    irrigation_method: str
    common_diseases: List[str]
    disease_prevention: str
    market_price_range: str


class WaterRequirementOut(BaseModel):
    crop_name: str
    water_requirement: str
    recommended_method: str


class AreaIntelligenceResponse(BaseModel):
    state: str
    district: str
    soil_report: SoilReportOut
    soil_improvement_recommendations: List[str]
    recommended_crops: List[CropRecommendationOut]
    water_guidance: List[WaterRequirementOut]
    generated_at: datetime
